libraries/AP_Airspeed/AP_Airspeed_NMEA.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ *   analog airspeed driver
+ */
+
+#include "AP_Airspeed.h"
+#include "AP_Airspeed_NMEA.h"
+
+extern const AP_HAL::HAL &hal;
+
+AP_Airspeed_NMEA::AP_Airspeed_NMEA(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+bool AP_Airspeed_NMEA::init()
+{
+    const AP_SerialManager& serial_manager = AP::serialmanager();
+
+    _uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_AirSpeed, 0);
+    if (_uart == nullptr) {
+        return false;
+    }
+
+    _uart->begin(serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_AirSpeed, 0));
+
+    set_use_zero_offset();
+    set_skip_cal();
+    set_offset(0);
+    set_ratio(1);
+    set_use(0);
```

Comment:
some explanation would be nice here
also, instead of using tube_order, a boolean function on backend that says if the sqrt should be applied, or asks for speed directly?

Reply:
fixed